RetailCoder.VBE/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs

```diff
@@ -0,0 +1,229 @@
+            var newLines = oldLines.Insert(firstReference.Selection.StartColumn - 1, GetDeclarationString(target));
+
+            _editor.DeleteLines(firstReference.Selection);
+            _editor.InsertLines(firstReference.Selection.StartLine, newLines);
+        }
+
+        private string GetDeclarationString(Declaration target)
+        {
+            return Environment.NewLine + "    Dim " + target.IdentifierName + " As " + target.AsTypeName + Environment.NewLine;
+        }
+
+        private void RemoveVariable(Declaration target)
+        {
+            Selection selection;
+            var declarationText = target.Context.GetText();
+            var multipleDeclarations = HasMultipleDeclarationsInStatement(target);
+
+            var variableStmtContext = GetVariableStmtContext(target);
+
+            if (!multipleDeclarations)
+            {
+                declarationText = variableStmtContext.GetText();
+                selection = GetVariableStmtContextSelection(target);
+            }
+            else
+            {
+                selection = new Selection(target.Context.Start.Line, target.Context.Start.Column,
+                    target.Context.Stop.Line, target.Context.Stop.Column);
+            }
+
+            var oldLines = _editor.GetLines(selection);
+
+            var newLines = oldLines.Replace(" _" + Environment.NewLine, string.Empty)
+                .Remove(selection.StartColumn, declarationText.Length);
+
+            if (multipleDeclarations)
+            {
+                selection = GetVariableStmtContextSelection(target);
+                newLines = RemoveExtraComma(_editor.GetLines(selection).Replace(oldLines, newLines));
+            }
+
+            _editor.DeleteLines(selection);
+
+            if (newLines.Trim() != string.Empty)
+            {
+                _editor.InsertLines(selection.StartLine, newLines);
+            }
+        }
+
+        private string RemoveExtraComma(string str)
+        {
+            if (str.Count(c => c == ',') == 1)
+            {
+                return str.Remove(str.IndexOf(','), 1);
+            }
+
+            var significantCharacterAfterComma = false;
+
+            for (var index = str.IndexOf("Dim", StringComparison.Ordinal) + 3; index < str.Length; index++)
+            {
+                if (!significantCharacterAfterComma && str[index] == ',')
+                {
+                    return str.Remove(index, 1);
+                }
+
+                if (!char.IsWhiteSpace(str[index]) && str[index] != '_' && str[index] != ',')
+                {
+                    significantCharacterAfterComma = true;
+                }
+
+                if (str[index] == ',')
+                {
+                    significantCharacterAfterComma = false;
+                }
+            }
+
+            return str.Remove(str.LastIndexOf(','), 1);
+        }
+
+        public Selection GetVariableStmtContextSelection(Declaration target)
```

Comment:
seems like a utility method that may be useful for other refactorings. did you consider extracting it to another class?